examples/helia-101/201-storage.js

```diff
@@ -1,50 +1,50 @@
 /* eslint-disable no-console */
-
+// @ts-check
+import { createHeliaHTTP } from '@helia/http'
 import { unixfs } from '@helia/unixfs'
 import { MemoryBlockstore } from 'blockstore-core'
-import { createHelia } from 'helia'
+import { FsBlockstore } from 'blockstore-fs'
 
 // the blockstore is where we store the blocks that make up files. this blockstore
 // stores everything in-memory - other blockstores are available:
 //   - https://www.npmjs.com/package/blockstore-fs - a filesystem blockstore (for use in node)
 //   - https://www.npmjs.com/package/blockstore-idb - an IndexDB blockstore (for use in browsers)
 //   - https://www.npmjs.com/package/blockstore-level - a LevelDB blockstore (for node or browsers,
 //                                        though storing files in a database is rarely a good idea)
-const blockstore = new MemoryBlockstore()
 
-// create a Helia node
-const helia = await createHelia({
-  blockstore
+// Create a new Helia node with an in-memory blockstore
+const helia1 = await createHeliaHTTP({
+  blockstore: new MemoryBlockstore()
 })
 
-// create a filesystem on top of Helia, in this case it's UnixFS
-const fs = unixfs(helia)
+// create a UnixFS filesystem on top of Helia
+const fs1 = unixfs(helia1)
 
 // we will use this TextEncoder to turn strings into Uint8Arrays
 const encoder = new TextEncoder()
 
+const message = 'Hello World 201'
+
 // add the bytes to your node and receive a unique content identifier
-const cid = await fs.addBytes(encoder.encode('Hello World 201'))
+const cid1 = await fs1.addBytes(encoder.encode(message))
 
-console.log('Added file:', cid.toString())
+console.log('Added file contents:', message)
 
-// create a second Helia node using the same blockstore
-const helia2 = await createHelia({
-  blockstore
+// Create a new Helia node with a filesystem blockstore
+const helia2 = await createHeliaHTTP({
+  blockstore: new FsBlockstore('./blockstore')
 })
 
-// create a second filesystem
 const fs2 = unixfs(helia2)
-
-// this decoder will turn Uint8Arrays into strings
-const decoder = new TextDecoder()
-let text = ''
-
-// read the file from the blockstore using the second Helia node
-for await (const chunk of fs2.cat(cid)) {
-  text += decoder.decode(chunk, {
-    stream: true
-  })
+try {
+  // Check if the CID is in the blockstore, which will be true if we ran this script before
+  const stats = await fs2.stat(cid1, { offline: true }) // `offline: true` will prevent the node from trying to fetch the block from the network
+  console.log(`Found ${cid1.toString()} in blockstore:`, stats)
+} catch (error) {
+  console.log("CID can't be found in the blockstore. We will add it now.")
+  // If the CID is not in the blockstore, we will add it now
+  const cid2 = await fs2.addBytes(encoder.encode(message))
+  console.log('Added file:', cid2.toString())
 }
 
-console.log('Added file contents:', text)
+process.exit(0)
```

Comment:
Do we need the `process.exit(0)` here?  If you stop the created helia nodes instead the process should end.

Reply:
i think this may be an artefact of 2color helping me bugtrap: on my WSL2/Ubuntu setup, `npm run` hangs and needs to be CTRL-C'd to kill the process? i still haven't been able to replicate it reliably/deduce which commands trigger it, or try different nodes with nvm, etc.  can switch to that later if you think there are enough WSL2 users to justify the bugtrapping session 😅  (Narrator: there weren't)